### PR TITLE
Introduce manual TARGET

### PR DESCRIPTION
**Pull request by grisu48**
When the TARGET is set to manual, the tester should take a given
CONTAINER_URL without modifying it.

### CODE REVIEW

**Review by dcermak — COMMENTED** [review]
This is not necessarily a good idea. With this change, you can now no longer test images built with both kiwi and docker in one run

**Comment by grisu48** [comment]
> This is not necessarily a good idea. With this change, you can now no longer test images built with both kiwi and docker in one run

We need this when setting custom image URLs that don't want to have auto-generated suffixes, see e.g. https://openqa.suse.de/tests/16000117#step/_root_BCI-tests_repository_/1

But I'm open to add a different target like `manual` and leave `custom` to how it works as of now.

**Comment by dirkmueller** [comment]
> But I'm open to add a different target like manual and leave custom to how it works as of now.

how about simply releasing sle15-image to ToTest? that would solve it without any code modifications on the qa pipeline.

**Comment by grisu48** [comment]
> > But I'm open to add a different target like manual and leave custom to how it works as of now.
> 
> how about simply releasing sle15-image to ToTest? that would solve it without any code modifications on the qa pipeline.

Yeah that's better here, but the ability to manually configure the container under test still appears like a good feature to have.

**Review by dcermak — COMMENTED** [review]
I am not entirely sold on this PR tbh. So far we have been using `baseurl` as the URL to the project where all your containers reside and let the test code figure out the correct url + build tag. With this change, you now have to set the url to the container directly, so the term `BASEURL` doesn't make a lot of sense anymore and it changes the existing behavior completely.

Have you considered using a different environment variable for the case `TARGET=manual`?

**Comment by grisu48** [comment]
> Have you considered using a different environment variable for the case `TARGET=manual`?

I now introduced the variable `CONTAINER_URL` to be used with `TARGET=manual`.

**Review by dcermak — DISMISSED** [review]

File: bci_tester/data.py [hunk 1]
```diff
@@ -350,7 +350,11 @@ def create_BCI(
     ):
         marks.append(pytest.mark.__getattr__(build_tag_base.replace(":", "_")))
 
-    if OS_VERSION == "tumbleweed":
+    if TARGET == "manual":
+        baseurl = os.getenv("CONTAINER_URL")
+        if not baseurl:
+            raise ValueError("Missing CONTAINER_URL for TARGET manual")
```
Comment (dirkmueller) [inline bci_tester/data.py:356]: this needs to be allowlisted in `tox.ini` . 

how is this different to TARGET=custom with BASEURL=xxx ?

Comment (grisu48) [reply bci_tester/data.py:356]: With `TARGET=custom` and `BASEURL=foobar` BCI-Test will still apply suffixes on the BASEURL.

**Review by dcermak — APPROVED** [review]

**Pull request merged**